app/Http/Controllers/GistsController.php

```diff
@@ -62,6 +67,32 @@ public function show($username, $gistId)
 
         return View::make('gistlogs.show')
             ->with('gistlog', $gistlog)
-            ->with('pageTitle', $gistlog->title.' | '.$gistlog->author);
+            ->with('pageTitle', $gistlog->title.' | '.$gistlog->author)
+            ->with('isStarredForUser', $this->isStarredForUser($gistId));
+    }
+
+    public function star(GistClient $client, $gistId)
+    {
+        $client->starGist($gistId);
+    }
+
+    public function unstar(GistClient $client, $gistId)
+    {
+        $client->unstarGist($gistId);
+    }
+
+    public function isStarredForUser($gistId)
+    {
+        try {
+            Auth::check();
+            $gistClient = app(GistClient::class);
+            $gistClient->github->authenticate(Auth::user()->token, GitHubClient::AUTH_HTTP_TOKEN);
```

Comment:
Ahh--so instead of making `github` public and doing this this way, could we add another method to the client that is this method? GistClient->isStarredForUser($gistId, $user) maybe?